lib/exec.ml

```diff
@@ -84,32 +92,38 @@ let ocaml_version ?ocamlc () =
   | Error (`Msg _) -> Error (`Msg "unable to find an installed ocamlc")
 
 let install_ocaml_to ~prefix ~src () =
-  OS.Dir.with_current src
-    (fun () ->
-      run_and_log Cmd.(v "./configure" % "--prefix" % p prefix) >>= fun () ->
-      run_and_log Cmd.(v "make" % "-j" % "world.opt") >>= fun () ->
-      run_and_log Cmd.(v "make" % "install"))
-    ()
-  >>= fun x -> x
+  let* x =
+    OS.Dir.with_current src
+      (fun () ->
+        let* () = run_and_log Cmd.(v "./configure" % "--prefix" % p prefix) in
+        let* () = run_and_log Cmd.(v "make" % "-j" % "world.opt") in
+        run_and_log Cmd.(v "make" % "install"))
+      ()
+  in
+  x
 
 let install_dune_to ~prefix ~src () =
-  OS.Dir.with_current src
-    (fun () ->
-      OS.File.write Fpath.(v "dune-workspace") "(lang dune 1.0)" >>= fun () ->
-      run_and_log Cmd.(v "ocaml" % "configure.ml" % "--libdir" % p prefix)
-      >>= fun () ->
-      run_and_log Cmd.(v "ocaml" % "bootstrap.ml") >>= fun () ->
-      run_and_log
-        Cmd.(
-          v "./dune.exe" % "build" % "-p" % "dune" % "--profile"
-          % "dune-bootstrap")
-      >>= fun () ->
-      run_and_log
-        Cmd.(
-          v "./dune.exe" % "install" % "--root" % p src % "--prefix" % p prefix
-          % "dune"))
-    ()
-  >>= fun x -> x
+  let* x =
```

Comment:
It sort of was, because it used `bind` without wrapping the result again so it merged nested `Ok Result.t`. As I was writing I felt a bit silly myself, so now I was motiveded to check whether there is an existing function, and turns out even the Stdlib comes with `Result.join` out-of-the-box, which is much better because it better conveys the intent instead of code that looks like someone forgot to do something with `x`.